core/lib/types/src/l2_to_l1_log.rs

```diff
@@ -44,6 +36,22 @@ impl L2ToL1Log {
         }
     }
 
+    /// Returns the minimum number of items in the Merkle tree built from L2-to-L1 logs
+    /// for a certain protocol version.
+    pub fn min_tree_size(protocol_version: ProtocolVersionId) -> usize {
```

Comment:
I'm not sure that it's the correct place for this function. 
 
separate functions `min_l2_l1_messages_tree_size` seems more appropriate

Reply:
L2L1Log knows nothing about tree

Reply:
Agree, moved to a separate function